src/whitenoise/storage/media_files.rs

```diff
@@ -0,0 +1,314 @@
+use crate::whitenoise::error::{Result, WhitenoiseError};
+use mdk_core::GroupId;
+use std::path::{Path, PathBuf};
+use tokio::fs;
+
+/// Filesystem storage for media files organized by MLS group
+///
+/// Directory structure:
+/// ```
+/// <cache_dir>/
+///   <mls_group_id_hex>/
+///     <subdirectory>/
+///       <filename>
```

Comment:
Just a reminder to update here as well if you take my comment regarding the scoping of files by group